### ISSUE TITLE
04_TranscriptionFactor_activity_with_Dorothea issue with run_viper

### ISSUE THREAD

**Issue by manas-kohli** [saezlab/transcriptutorial#46]
I'm trying to use the function run_viper to calculate transcription factor activity based on an earlier differential expression analysis I ran but running the following line:

tf_activities_stat <- 
  dorothea::run_viper(ttopKOvsWT_matrix, regulons,
                      options =  list(minsize = 4, eset.filter = FALSE, 
                                      cores = 1, verbose = TRUE, method = c("scale")))

returns a vector of NA values. I also get the following error message:
Warning message:
In FUN(newX[, i], ...) : no non-missing arguments to max; returning -Inf

I then tried to follow the example provided in the documentation but even with your data I run into the same error. I did preprocess the differential expression matrix to keep only the t values and gene names as row names as you guys had performed in your example in the vignette. run_viper also works fine on a matrix of vst transformed counts but I can't seem to get it to work only on the differential expression results. I'm wondering if anyone else has the same issue and can reproduce the error since I seem to be getting it on the example data as well.

**Comment by adugourd** [comment]
Yes, I'm assuming the function breaks because a single column is passed. To solve this in a immediate manner, you may try to dupplicate your column and pass a dataframe with 2 identical column as input.

In the meantime, the best current tutorial to get TF activities can be found here, in the updated decoupleR package documentation:

https://saezlab.github.io/decoupleR/articles/tf_bk.html

**Comment by manas-kohli** [comment]
Thank you for your response! I managed to implement decoupleR to estimate TF activities using the differential expression results. I want to pass this to carnival now but the Rmds in this repo seem to have disappeared. Will they be brought back since particularly the visualisation Rmd of carnival results was quite useful and I would like to be able to follow the entire workflow unless this is already somewhere and I'm just missing the link

**Comment by adugourd** [comment]
Yes, I did some clean up because this tutorial was starting to become quite deprecated.

The new tutorial there to run carnival should be more up to date. I will see if I can add back the function to visualise the network in R.

However, I would strongly advise to use cytoscape for visualisation of the network output of carnival in the futur. It has nice graphical user interface and will allow much easier exploration of the networks.

I will also readd the network exploration with pathway analysis later, when it is updated to our current best practice.

**Comment by adugourd** [comment]
I added back the network visualisation function

**Comment by gabora** [comment]
Hi, 

very strange error you get. My first guess is that there is an issue with the space and brackets in the path to the files. Could you move your working directory out of Dropbox to a location without space and bracets? 

If that did not work, then please try to run some of our tests which uses small models. 
you can clone the repo and then run the test in the `CARNIVAL/tests/testthat` folder. 
Look for `test_cbc_small_models.R`, test_gurobi_small_models.R` and  `test_cplex_small_models.R` files. 
In the tests, make sure that they finds your Gurobi/CBC/CPLEX installation.

**Comment by manas-kohli** [comment]
Thank you for your reply. It did seem to be a path issue and when I copied it to another folder it works fine with the cbc solver. I still have troubles with Gurobi and get this specific error:

C:\GUROBI~2\win64\bin\\..\python37\bin\python: can't open file 'MIPGAP=0': [Errno 2] No such file or directory

I'm not sure what's wrong but CARNIVAL appears to be working with cbc so I can still proceed with my analysis
